ls1intum/Artemis
# `General`: Improve chart appearance in course statistics page

### Checklist
#### General


#### Client

### Motivation and Context


This PR improves the charts on students statistics tab in some details

### Description


- **Chart coloring**: The coloring of the bars is reworked. A bit darker colors are chosen since the current ones are quite loud. For this purpose, standardized colors used in different places are reused. Furthermore, the color of the average score line is changed from red to yellow. Since red is used to indicate missing scores in the bar charts, this leads to a confusing difference in color coding between the two chart types.
- **Performance in exercises positioning**: The positioning of the chart is now changed so that it is centered on the page and consumes 90% of the page width.
- **Readability of legend entries**: The default color of the legend entry test is changed to the Artemis dark color (which is more or less black) as well as its font weight is increased in order to make the legend more readable. If the user now hovers over a chart entry, it is colored in Artemis primary color instead of black as it was previously.
- **Horizontal bar chart height**: We encountered some problems for the horizontal bar charts: As the height of the chart is currently hardcoded, the chart gets hardly readable if many exercises should be displayed. The height is now determined dynamically in order to ensure readability even if many exercises are displayed.

### Steps for Testing

Prerequisites:
- 1 Student
- 1 Course with some contributions

1. Log in to Artemis
2. Navigate to the student statistics tab
3. Inspect the charts there. Feel free to share your opinion about the new colors. These are open for discussion.
4. Also check that the horizontal bar charts are sized conveniently and the dimensions are not reducing your usability.

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage





Mainly UI changes

### Screenshots


An example how a chart with 50 exercises would look with the current implementation. Notice that even some exercise titles are omitted by the framework because there is not enough space:
![Old_impl](https://user-images.githubusercontent.com/80622272/154852119-a1a483d0-d0a6-4197-8929-60203f7f93f3.PNG)

How it looks with my changes:
![image](https://user-images.githubusercontent.com/80622272/154852145-7ecd1be1-e957-4095-8ac4-54e56d6176d0.png)

The new colors:
![image](https://user-images.githubusercontent.com/80622272/154852457-30767289-300a-4caa-885f-edd7a1f45de2.png)
